public/app/app.ts

```diff
@@ -161,5 +169,20 @@ function addClassIfNoOverlayScrollbar() {
     document.body.classList.add('no-overlay-scrollbar');
   }
 }
-
+async function resolveHomePage() {
+  const homePage: string = config.homePage;
+  const userPrefs = await UserPreferencesService.load();
+  const baseUrl = config.appSubUrl || '';
+  const isHomePage = locationUtil.stripBaseFromUrl(window.location.pathname) === '/';
+
+  // When user hasn't configured home dashboard and the config has home page settings -> redirect to that page
+  if (userPrefs.homeDashboardId === 0 && Boolean(homePage)) {
+    const newUrl = locationUtil.stripBaseFromUrl(homePage);
+    config.homeUrl = `${baseUrl}${newUrl}`;
+
+    if (isHomePage) {
+      locationService.replace(newUrl);
+    }
+  }
+}
```

Comment:
I think we can move this to a `/` home route instead  (defined in routes.ts) that loads a HomePage.tsx component that handles this logic. feels cleaner than doing during app init. 

We can then move the default home dashboard route to `/dashboard/home`